filebeat/crawler/prospector.go

```diff
@@ -35,9 +36,42 @@ func (p *Prospector) Init() error {
 		return err
 	}
 
+	var prospectorer Prospectorer
+
+	switch p.ProspectorConfig.Harvester.InputType {
+	case cfg.StdinInputType:
+		prospectorer = ProspectorStdin{
+			Prospector: p,
```

Comment:
I moved iterationg and lastscan to log, but running and registrar are used by both. I actually prefer to pass the full object instead of multiple variables as there are less variables around. Do you see here some disadvantages?